docs/formats/object-detection/custom.md

```diff
@@ -0,0 +1,96 @@
+# Custom Object Detection Format
+
+## Overview
+The Custom format provides an in-memory representation for object detection labels. It's designed for programmatic creation and manipulation of label data within Python scripts. This format is particularly useful when you need to convert labels generated by custom models or dataloaders into a standardized structure before potentially converting them to other file-based formats supported by `labelformat`.
+
+## Specification of Custom Detection Format
+The format is defined by the `CustomObjectDetectionInput` dataclass in `labelformat.formats.custom`. It holds the following information directly in Python objects:
+
+- `categories`: A list of `Category` objects, each containing `id` and `name`.
+- `images`: A list of `Image` objects, each containing `id`, `filename`, `width`, and `height`.
+- `labels`: A list of `ImageObjectDetection` objects. Each object links an `Image` to a list of `SingleObjectDetection` instances.
+  - `SingleObjectDetection` contains:
+    - `category`: The `Category` object for the detection.
+    - `box`: A `BoundingBox` object representing the location (can be created from various formats like XYXY, XYWH, CXCYWH).
+    - `confidence`: An optional float score (0-1).
+
+**Note:** This format is primarily intended for programmatic use and does not support direct loading from files via the CLI using `add_cli_arguments`. It serves as a flexible in-memory structure.
+
+## Example Usage (Programmatic Creation)
+```python
+from labelformat.formats.custom import CustomObjectDetectionInput
+from labelformat.model.bounding_box import BoundingBox, BoundingBoxFormat
+from labelformat.model.category import Category
+from labelformat.model.image import Image
+from labelformat.model.object_detection import (
+    ImageObjectDetection,
+    SingleObjectDetection,
+)
+
+# Assume you have:
+# my_dataloader: An iterable yielding (PILImage, filename)
+# my_model: A model with a .predict() method and .get_categories()
+# prediction_bbox_format: The format of your model's output boxes ("xyxy", "xywh", etc.)
+
+categories = [
+    Category(id=i, name=category_name)
+    for i, category_name in enumerate(my_model.get_categories())
+]
+category_map = {cat.name: cat for cat in categories} # Or map by index if model outputs index
+
+images = []
+labels = []
+image_id_counter = 0
+
+for pil_image, filename in my_dataloader:
+    # Create Image object
+    current_image = Image(
+        id=image_id_counter,
```

Comment:
Do we need the id here? What is it for?

Reply:
It is a required attribute of the labelformat class Image.
If we wanted to change that, we would need to rework many other parts of the code, which belongs to a different PR.